UDPServer/SiceUDPServer.cs

```diff
@@ -78,6 +91,51 @@ public async Task GetData()
             }
         }
 
+        public async Task GetLogin()
+        {
+            var remoteIpEndPoint = new IPEndPoint(ConnectionIP, ConnectionPort);
+            var data = await client.ReceiveAsync();
+            ReceivedData = Encoding.ASCII.GetString(data.Buffer);
+            //var credential = JsonSerializer.Deserialize(ReceivedData, ClientLoginEvent, default);
+            var credential = JsonConvert.DeserializeObject<ClientLoginEvent>(ReceivedData);
+            string username = credential.Username;
+            string password = credential.Password;
+            using (var db = new ServerContext())
+            {
+                var query = from login in db.Login
+                            where login.Username == username
+                            select login;
+                foreach (var items in query)
+                {
+                    if(items.Username == username && BCrypt.Net.BCrypt.Verify(password, items.PasswordHash))
+                    {
+                        HasLoggedIn = true;
+                        ConnectedControllerID = items.ControllerID;
+                        GetControllerInfo(ConnectedControllerID);
+                        eventAggregator.PublishOnUIThread(new ServerStatusChangedEvent(ServerStatus.Listening, ConnectedControllerInfo));
+                    }
+                }
+            }
+        }
+
+        public void GetControllerInfo(int controllerID)
+        {
+            using (var db = new ServerContext())
+            {
+                var query = from controller in db.Controllers
+                            where controller.ControllerID == controllerID
+                            select controller;
+                foreach (var item in query)
+                {
+                    if(item.ControllerID == controllerID)
+                    {
+                        this.ConnectedControllerInfo = item.ControllerInfo;
+                    }
+                }
+
+            }
+        }
```

Comment:
Same as above. The ServerContext should be initialised through the DI container.